src/segment_brewfather.go

```diff
@@ -0,0 +1,341 @@
+	bf.StatusIcon = bf.getBatchStatusIcon(data.Status)
+
+	fermStartDate := time.UnixMilli(bf.Batch.FermentStartDate)
+	bottlingDate := time.UnixMilli(bf.Batch.BottlingDate)
+
+	switch bf.Batch.Status {
+	case BFStatusFermenting:
+		// in the fermenter now, so relative to today.
+		bf.DaysFermenting = uint(time.Since(fermStartDate).Hours() / 24)
+		bf.DaysBottled = 0
+		bf.DaysBottledOrFermented = &bf.DaysFermenting
+	case BFStatusConditioning, BFStatusCompleted, BFStatusArchived:
+		bf.DaysFermenting = uint(bottlingDate.Sub(fermStartDate).Hours() / 24)
+		bf.DaysBottled = uint(time.Since(bottlingDate).Hours() / 24)
+		bf.DaysBottledOrFermented = &bf.DaysBottled
+	default:
+		bf.DaysFermenting = 0
+		bf.DaysBottled = 0
+		bf.DaysBottledOrFermented = nil
+	}
+
+	// URL property set to weblink to the full batch page
+	batchID := bf.props.getString(BFBatchID, "")
+	if len(batchID) > 0 {
+		bf.URL = fmt.Sprintf("https://web.brewfather.app/tabs/batches/batch/%s", batchID)
+	}
+
+	bf.DefaultString = bf.formatDefaultString()
+
+	return true
+}
+
+// build a default string for the segment using Sprintf for speed and to provide a simple starting point without complex template logic
+func (bf *brewfather) formatDefaultString() string {
+	dayIcon := bf.props.getString(BFDayIcon, "d")
+	switch bf.Batch.Status {
+	case BFStatusPlanning, BFStatusBrewing:
+		return fmt.Sprintf("[%s %s %.1f%%](%s)", bf.StatusIcon, bf.Recipe.Name, bf.MeasuredAbv, bf.URL)
+	case BFStatusFermenting:
+		ret := fmt.Sprintf("[%s %d%s %s %.1f%%", bf.StatusIcon, bf.DaysFermenting, dayIcon, bf.Recipe.Name, bf.MeasuredAbv)
+		if bf.Reading != nil {
+			ret += fmt.Sprintf(" %1.3f %.1f° %s", bf.Reading.Gravity, bf.Reading.Temperature, bf.TemperatureTrendIcon)
+		}
+		ret += "](" + bf.URL
+		ret += ")"
+		return ret
+	case BFStatusConditioning, BFStatusCompleted, BFStatusArchived:
+		return fmt.Sprintf("[%s %d%s %s %.1f%%](%s)", bf.StatusIcon, bf.DaysBottled, dayIcon, bf.Recipe.Name, bf.MeasuredAbv, bf.URL)
+	}
+
+	return ""
+}
+
+func (bf *brewfather) getTrendIcon(trend float64) string {
+	// Not a fan of this logic - wondering if Go lets us do something cleaner...
+	if trend >= 0 {
+		if trend > 4 {
+			return bf.props.getString(BFDoubleUpIcon, "↑↑")
+		}
+
+		if trend > 2 {
+			return bf.props.getString(BFSingleUpIcon, "↑")
+		}
+
+		if trend > 0.5 {
+			return bf.props.getString(BFFortyFiveUpIcon, "↗")
+		}
+
+		return bf.props.getString(BFFlatIcon, "→")
+	}
+
+	if trend < -4 {
+		return bf.props.getString(BFDoubleDownIcon, "↓↓")
+	}
+
+	if trend < -2 {
+		return bf.props.getString(BFSingleDownIcon, "↓")
+	}
+
+	if trend < -0.5 {
+		return bf.props.getString(BFFortyFiveDownIcon, "↘")
+	}
+
+	return bf.props.getString(BFFlatIcon, "→")
+}
+
+func (bf *brewfather) getBatchStatusIcon(batchStatus string) string {
+	switch batchStatus {
+	case BFStatusPlanning:
+		return bf.props.getString(BFPlanningStatusIcon, "")
+	case BFStatusBrewing:
+		return bf.props.getString(BFBrewingStatusIcon, "")
```

Comment:
the defaults need the `\uxxxx` notation

Reply:
That makes more sense than the evil glyph boxes of doom!

Reply:
on that topic - I notice some spacing misbehaviours depending on the glyph, are you tracking issues on that already?